### ISSUE TITLE
Is it possible to change colors of VCF track according to the values in a specific column?

### ISSUE THREAD

**Issue by diogokawamoto** [igvteam/igv.js#1217]
Hi,

I am currently working with structural variants and I have some VCF files that have multiple SVs events in the same region, I would like to have some visual way to easily differentiate the events by its type (DEL, DUP etc).

Is there any setting to change the color of VCF tracks according to its 'ALT' value or something like this?

Thank you in advance.

**Comment by jrobinso** [comment]
Currently it is not, although its a good suggestion and should be implemented.   I will leave this open as a suggestion.

If you are a javascript programmer you could modify "js/variant/variantTrack.js" , lines 198-203 for the variant and 229-237 for the call, to accomplish this until we implement it.

**Comment by diogokawamoto** [comment]
Thank you! I appreciate your attention!

**Comment by haoyuanli** [comment]
> Thank you! I appreciate your attention!

@diogokawamoto 

Random question, do you just host your VCF files on some static hosting server, like aws?

**Comment by jrobinso** [comment]
@haoyuanli Mostly AWS, yes,  but sometimes Dropbox and Google, as well as an apache server on our own site.  But always static hosting.

**Comment by jrobinso** [comment]
@diogokawamoto Since you are working with VCFs containain SVs and I'm not could you provide me a list of types for colors (DEL, DUP, what else???).   Also feel free to suggest colors.   We could use just random colors from a pallete.

**Comment by jrobinso** [comment]
@haoyuanli See this link for tips on setting up a static server for use with javascript.  In particular you have to do something about CORS,  not these solutions necessarily but it has to be dealt with  https://github.com/igvteam/igv.js/wiki/Data-Server-Requirements

**Comment by haoyuanli** [comment]
> @haoyuanli See this link for tips on setting up a static server for use with javascript. In particular you have to do something about CORS, not these solutions necessarily but it has to be dealt with https://github.com/igvteam/igv.js/wiki/Data-Server-Requirements

Thanks! I will give those a read.

**Comment by diogokawamoto** [comment]
> @diogokawamoto Since you are working with VCFs containing SVs and I'm not could you provide me a list of types for colors (DEL, DUP, what else???). Also feel free to suggest colors. We could use just random colors from a pallete.

@jrobinso sorry for the delayed response. Well...

According to the [spec](https://samtools.github.io/hts-specs/VCFv4.3.pdf)
The SVTYPE should be one of the following values: DEL, INS, DUP, INV, CNV, BND.

For the color palette, I don’t know if there is a official standard, but in the following [website](https://www.ncbi.nlm.nih.gov/dbvar/content/overview/) in the section ‘dbVar Variant rendering’ we can find the colours they are using.

And if you need some VCFs files to test, I’m current using some from this [server](https://ftp.ncbi.nlm.nih.gov/pub/dbVar/data/Homo_sapiens/by_study/vcf/) of NCBI, for example: 
vcf: https://ftp.ncbi.nlm.nih.gov/pub/dbVar/data/Homo_sapiens/by_study/vcf/nstd186.GRCh38.variant_call.vcf.gz
index: https://ftp.ncbi.nlm.nih.gov/pub/dbVar/data/Homo_sapiens/by_study/vcf/nstd186.GRCh38.variant_call.vcf.gz.tbi

Thank you for your attention.

**Comment by jrobinso** [comment]
I'm working on this now.   The request is a bit ambiguous as there are 2 types of objects displayed in a VCF track:  variants and genotypes (i.e. samples or calls).   SVTYPE is an info field for the variant, so I am working on that now.

**Comment by jrobinso** [comment]
Here's a very general way for a developer to set a custom color based on any info field using a color function.   I will work next on a  solution for end users through the track menu, but this is more difficult to handle in a general way since info fields can have many types.   Perhaps the menu will be restricted to "string" info fields.

Note support for this hasn't been released yet

```
    const options =
        {
            locus: "chr22",
            genome: "hg38",
            tracks: [
                {
                    url: "https://s3.amazonaws.com/igv.org.demo/nstd186.GRCh38.variant_call.vcf.gz",
                    indexURL: "https://s3.amazonaws.com/igv.org.demo/nstd186.GRCh38.variant_call.vcf.gz.tbi",
                    name: "nstd186",
                    visibilityWindow: -1,
                    color: function (variant) {
                        const svtype = variant.info["SVTYPE"];
                        switch (svtype) {
                            case 'DEL':
                                return "#ff2101";
                            case 'INS':
                                return "#001888";
                            case 'DUP':
                                return "#028401";
                            case 'INV':
                                return "#008688";
                            case 'CNV':
                                return "#8931ff";
                            case 'BND':
                                return "#891100";
                            default:
                                return "#002eff";
                        }
                    }
                }
            ]
        };
```

**Comment by jrobinso** [comment]
This is released now with 2.8.0
